exporter/loadbalancingexporter/exporter.go

```diff
@@ -120,7 +122,7 @@ func (e *exporterImp) Start(ctx context.Context, host component.Host) error {
 }
 
 func (e *exporterImp) onBackendChanges(resolved []string) {
-	resolved = sort.StringSlice(resolved)
```

Comment:
Are you seeing this list unsorted? This should be sorted by the resolver already, so, this call here is just for local safety. In any case, we shouldn't be changing the `resolved` argument.

Reply:
What was the purpose of the line I changed? It is not either sort of the strings and not a check that it is sorted. This line literally does nothing at the moment

Reply:
Another 🤦🏽, it's casting to the type `StringSlice` instead of actually calling a function that sorts the slice... 

Is the sort required here at all? This list should be sorted already by the backend anyway...

Reply:
It is sorted in dns resolver but not in static resolver. I would add sort for static resolver and log error here if the slice isn't sorted, because this can lead to some problems that multiple loadbalancers (why not) are not synced in terms of endpoints

Reply:
By the way it's a nice feature that the hash is consistent among instances. Given that, there is a real possibility to move loadbalancer to agents instead of putting it to a separate collector. In my opinion it is not hard to support this useful behaviour

Reply:
We sort the backends in the DNS resolver because it's easier to determine whether the list changed. One edge case that it solves is that two endpoints resolving to the same angle in the ring would cause undetermined behavior on unsorted lists, so that updates to the same list could yield different results for the same angle. With a sorted list, we have a consistent view in this case as well.

For static backends, we don't need this, as the list is never changing, the view of the ring is always the same.

Reply:
If you don't believe, that one can do a mistake by using several exporters with different order of hosts in static resolver config, then I totally agree with you. Otherwise putting one line for sort not much of an issue to do in this PR

Reply:
I'd rather have this in a different PR. Keeping one PR per problem is good for sanity :-)

Reply:
At least I think that the order of hosts in static resolver config shouldn't affect the behaviour of exporter in any way

Reply:
Sure, I'll move these changes to a separate PR